### ISSUE TITLE
Steam crash on startup in fedora 26.

### ISSUE THREAD

**Issue by samurouse** [ValveSoftware/steam-for-linux#5112]
#### Your system information

* Steam client version (build number or date): 1501542808
* Distribution (e.g. Ubuntu): Fedora 26
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Steam crashes as I try to log in. 
It always open the usual "checking for updates" then "Connecting into account", then it crashes. Sometimes steam ask for my password and an steam guard code sent to my phone, then after I input this info, it crashes.

Opening a terminal and running steam prints the following to stdout/stderr:
[stdout_steam_crash_output.txt](https://gist.github.com/samurouse/ca2c260afc4c8e455772ba36a6bd0463)

In the above Gist there are two assertion failures reported:
    m_AssocHandles.IsValidIndex( nAssocIdx ) ):completionportmanager_posix.cpp:629
    ../common/pipes.cpp (572) : Assertion Failed: m_pInternalPipe->BRead failed

And three core dumps, two of them marked as uploaded. I'm attaching all three of them for the sake of completeness.
Finally, I have removed app id and Steam id information from the above Gist for privacy reasons.

#### Steps for reproducing this issue:

1. Boot in fedora 26
2. Open steam

[assert_20170801214348_1.dmp.txt](https://github.com/ValveSoftware/steam-for-linux/files/1192355/assert_20170801214348_1.dmp.txt)
[crash_20170801214400_26.dmp.txt](https://github.com/ValveSoftware/steam-for-linux/files/1192356/crash_20170801214400_26.dmp.txt)
[crash_20170801214412_2.dmp.txt](https://github.com/ValveSoftware/steam-for-linux/files/1192357/crash_20170801214412_2.dmp.txt)

**Comment by kisak-valve** [comment]
Hello @samurouse, this issue is already being tracked at #5101. You should be able to pin the steam runtime variant of dbus with `ln -s ${HOME}/.local/share/Steam/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu/libdbus-1.so.3 ${HOME}/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbus-1.so.3` to workaround this issue.

Closing as a duplicate.

**Comment by samurouse** [comment]
The suggested workaround did not work for me. There is no "Steam" folder in "${HOME}/.local/share/" for my installation.
Steam updated today but the issue is still happening

**Comment by lig** [comment]
```
ln -s $STEAM_DIR/ubuntu12_32/steam-runtime/{i386/lib/i386-linux-gnu,pinned_libs_32}/libdbus-1.so.3
ln -s $STEAM_DIR/ubuntu12_32/steam-runtime/{amd64/lib/x86_64-linux-gnu,pinned_libs_64}/libdbus-1.so.3
```
Worked for me

Usually `$HOME/.steam/root` is a symlink to your Steam dir.

**Comment by samurouse** [comment]
@lig should I just copy and paste these? It didn't worked for me either.
It complains about the paths between curly brackets.

**Comment by samurouse** [comment]
@kisak-valve it seems that the issue indeed was similar to #5101. Based on the workaround you and @lig mentioned, I was able to make it work with the following line:

ln -s ~/Steam/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu/libdbus-1.so.3 ~/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/

After this I launched steam and it asked my password again, but this time it it actually opened after I put the password and steam guard code!
I feel a bit dumb for not trying to craft this command earlier, but I had little time in the previous days and wanted some copy-and-paste solution.
Thanks for the help.
